modules/tournament/src/main/TournamentScheduler.scala

```diff
@@ -390,13 +392,13 @@ Thank you all, you rock!"""
   def receive = {
 
     case ScheduleNow =>
-      TournamentRepo.scheduledUnfinished.map(_.flatMap(_.schedule)) map
-        ScheduleNowWith.apply pipeTo self
+      TournamentRepo.scheduledUnfinished map { lst =>
+        self ! ScheduleNowWith(lst.filter(_.schedule.nonEmpty))
```

Comment:
`scheduledUnfinished` should only return non empty schedules

Reply:
I’m not sure what you mean :-(

Reply:
I assume he means that .nonEmpty is useless because there will never be empty schedules.

Reply:
I agree a `Tournament.WithSchedule(tour, sched)` datatype is best

Reply:
Yes that might work too. 

Downside of outside case class is that two schedule objects for the same tourney are possible, and you can’t put unscheduled and scheduled tourneys together in a collection if you want to later see schedules.

Reply:
Btw I’m not planning on changing this for the PR, was just commenting about it in general.